microsoft/testsuites/firewalld/firewalldsuite.py

```diff
@@ -0,0 +1,182 @@
+    # - overall status (PASSED/FAILED)
+    ls = node.tools[Ls]
+    if not ls.path_exists(log_file):
+        raise LisaException(
+            f"Consolidated firewalld test log path: {log_file} doesn't exist, "
+            "please check if testing runs well or not."
+        )
+
+    cmd_total = f"cat {log_file} | grep \"tests were run\" | cut -d' ' -f 2"
+    total_tests = node.execute(cmd_total, sudo=True, shell=True)
+
+    cmd_fail = f"cat {log_file} | grep \"failed unexpectedly\" | cut -d' ' -f 1"
+    tests_failed = node.execute(cmd_fail, sudo=True, shell=True)
+
+    cmd_skip = f"cat {log_file} | grep \"were skipped\" | cut -d' ' -f 1"
+    tests_skipped = node.execute(cmd_skip, sudo=True, shell=True)
+
+    # Logs dir contains the list of failed tests.
+    fail_testcase_num: List[str] = []
+    failed_tests_list = ls.list_dir(log_dir, sudo=True)
+    for test in failed_tests_list:
+        fail_testcase_num.append(test.split("/")[-2])
+
+    # Note: There are known failures due to missing support for `FIB`
+    # based expressions in kernel.
+    # This is a temporary known (and constant) list of failures
+    known_fail_testcase_num = [
+        "061",
+        "107",
+        "120",
+        "124",
+        "192",
+        "193",
+        "194",
+        "195",
+        "196",
+        "197",
+        "200",
+        "240",
+        "252",
+        "306",
+        "324",
+    ]
+
+    status = TestStatus.PASSED
+    if known_fail_testcase_num != fail_testcase_num:
+        status = TestStatus.FAILED
+
+    # Update the test result data
+    result.set_status(
+        status,
+        f"TOTAL:{total_tests}\nFAILED:{tests_failed}\nSKIPPED:{tests_skipped}\n",
+    )
+
+
+@TestSuiteMetadata(
+    area="firewalld",
+    category="functional",
+    description="""
+    This test suite is to validate firewalld daemon on Azure Linux 3.0.
+    """,
+)
+class FirewalldSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+        if not isinstance(node.os, CBLMariner) or node.os.information.version < "3.0.0":
+            raise SkippedException(
+                UnsupportedDistroException(
+                    node.os, "Firewalld testsuite is supported only on AzureLinux 3.0."
+                )
+            )
+
+    @TestCaseMetadata(
+        description="""
+        This test runs the firewalld testsuite.
+        The testsuite is provided by the firewalld-test rpm.
+        """,
+        priority=3,
+    )
+    def verify_firewalld(
+        self,
+        node: Node,
+        log: Logger,
+        result: TestResult,
+    ) -> None:
+        # Install the dependencies for running the test suite
+        if isinstance(node.os, CBLMariner):
+            node.os.install_packages(
+                [
+                    "ebtables",
+                    "nftables",
+                    "python3-dbus",
+                    "iproute",
+                    "firewalld",
```

Comment:
Is the firewalld the default firewall on Azure Linux 3.0? If so, it doesn't need to install here. If not, is there any reason to test it on Azure Linux?